pipelinewise/fastsync/commons/tap_mongodb.py

```diff
@@ -0,0 +1,271 @@
+                    utils.SDC_DELETED_AT: None
+                }
+
+                writer.writerow(row)
+
+                exported_rows += 1
+
+        os.remove(export_file_path)
+        LOGGER.info('Exported total of %s rows from %s...', exported_rows, table_name)
+
+    @staticmethod
+    def _get_collection_columns() -> Tuple:
+        """
+        Get predefined table/collection column details
+        """
+        return (
+            ('_ID', 'string'),
+            ('DOCUMENT', 'object'),
+            (utils.SDC_EXTRACTED_AT, 'datetime'),
+            (utils.SDC_BATCHED_AT, 'datetime'),
+            (utils.SDC_DELETED_AT, 'string'),
+        )
+
+    def fetch_current_log_pos(self)->Dict:
+        """
+        Find and returns the latest ChangeStream token.
+        LOG_BASED method uses changes streams.
+        MongoDB doesn't have any built-in feature to get the most recent token atm,
+        so a workaround is to start a cursor, grab the first token it returns then exit.
+
+        Returns: token
+
+        """
+        token = None
+
+        with self.database.watch(max_await_time_ms=1000) as cursor:
+            while cursor.alive:
+                _ = cursor.try_next()
+                token = cursor.resume_token
+
+                if token is not None:
+                    break
+
+        return {
+            'token': token
+        }
+
+    # pylint: disable=invalid-name
+    def fetch_current_incremental_key_pos(self, fully_qualified_table_name: str, replication_key: str):
+        """
+        No Implemented
+        Args:
+            fully_qualified_table_name:
+            replication_key:
+        """
+        raise NotImplementedError('INCREMENTAL method is not supported for tap-mongodb')
+
+    def map_column_types_to_target(self):
+        """
+        Create a map of columns and their target type in addition of primary keys
+        Returns: dictionary
+
+        """
+        mapped_columns = []
+
+        for column_name, column_type in self._get_collection_columns():
+            mapped_columns.append(f'{column_name} {self.tap_type_to_target_type(column_type)}')
+
+        return {
+            'columns': mapped_columns,
+            'primary_key': ['_ID']
+        }
+
+    def _export_collection(self, export_dir: str, collection_name)->str:
+        """
+        Dump a collection data into a compressed bson file and returns the path
+        Args:
+            export_dir: Specifies the directory where dumped file will be
+            collection_name: Name of the collection to dump
+
+        Returns: Path to the file
+
+        """
+        url = f'mongodb://{self.connection_config["user"]}:{self.connection_config["password"]}' \
+              f'@{self.connection_config["host"]}:{self.connection_config["port"]}/' \
+              f'{self.connection_config["database"]}?authSource={self.connection_config["auth_database"]}'
+
+        return_code = subprocess.call([
+            'mongodump',
```

Comment:
can we install mongodump binary to the [Dockerfile](https://github.com/transferwise/pipelinewise/blob/9fd886038cc19ed62ea200ba1f4f5d72275624c4/Dockerfile#L6) as well similar to the dev-env [here](https://github.com/transferwise/pipelinewise/blob/9fd886038cc19ed62ea200ba1f4f5d72275624c4/dev-project/entrypoint.sh#L5) ?

Reply:
Oh, I missed that one! will do, thanks.

Reply:
Fixed in commit https://github.com/transferwise/pipelinewise/pull/434/commits/ffa04a5233ae1ca4c46e9a97787ef3d466ff1e8a